### ISSUE TITLE
Svelte 5 Pre-release Compatibility

### ISSUE THREAD

**Issue by EMH333** [EMH333/esbuild-svelte#235]
This is different than #234 since npm does some funny things with version locks and the prerelease versions may have subtle changes compared to the actual release.

This is a good time to clarify my commitment to prerelease support:
1. I promise that `esbuild-svelte` will have an installable version compatible with Svelte 5 the day it is released. It may not support all the new features, but it will at least let projects compile with Svelte 5. Worth noting that I am intending `v0.8.1`  to be that version, but won't hesitate to make changes as required. 
2. Support for prereleases is best effort only and will stop immediately (and retroactively) once a production version of Svelte 5 is released.
3. The usual DX does not exist for prerelease versions (see below) since they are, well, prerelease. Hacky workarounds may be required, with the expectation that they go away on the release of Svelte 5.

**Comment by EMH333** [comment]
> I'm not sure if I'm doing something wrong, but I'm seeing this when I run `npm install`.
> 
> ```
> $ npm install
> npm ERR! code ERESOLVE
> npm ERR! ERESOLVE unable to resolve dependency tree
> npm ERR! 
> npm ERR! While resolving: project@undefined
> npm ERR! Found: svelte@5.0.0-next.136
> npm ERR! node_modules/svelte
> npm ERR!   dev svelte@"^5.0.0-next.136" from the root project
> npm ERR! 
> npm ERR! Could not resolve dependency:
> npm ERR! peer svelte@">=3.43.0 <6" from esbuild-svelte@0.8.1
> npm ERR! node_modules/esbuild-svelte
> npm ERR!   dev esbuild-svelte@"^0.8.1" from the root project
> ```
> 
> My `package.json` looks like this.
> 
> ```
> {
>   "name": "project",
>   "devDependencies": {
>     "esbuild": "^0.21.0",
>     "esbuild-svelte": "^0.8.1",
>     "svelte": "^5.0.0-next.136"
>   }
> }
> ```
> 
> I updated to the latest `esbuild-svelte` and installed `svelte@next`.

**Comment by EMH333** [comment]
TL;DR: Use `--force` or `--legacy-peer-deps` for the prereleases since there isn't a great way to allow peer dependencies of all prerelease versions of a package.

Essentially, semver doesn't consider prereleases of v2 to be `<3`. As a result, even though the human-readable version number of `2.0.0-beta.1` is clearly less than `v3`, semver doesn't consider that to be the case. This means that dependency graph solving fails.

There are numerous issues about this (https://github.com/npm/cli/issues/2087 as an example) and an active RFC on how to resolve this https://github.com/npm/rfcs/pull/397.

But I don't think there is a clear/easy way to fix this for the prerelease versions. This won't be an issue once Svelte 5 is officially released.

**Comment by EMH333** [comment]
Interesting! Let me play around with that and see what I can do

**Comment by EMH333** [comment]
The change to allow Svelte 5 pre-releases as a peer dependency has been released as `v0.8.2`. Sorry for the delay!

**Comment by EMH333** [comment]
[Svelte 5 has officially been released](https://github.com/sveltejs/svelte/releases/tag/svelte%405.0.0), meaning esbuild-svelte will no longer provide support or compatibility for prereleases. Closing this issue.
